### PR TITLE
BAEL-7206

### PR DESCRIPTION
**Pull request by shahulbasha**
BAEL-7206 Created a new module as there was no module specific to maven-build-lifecycle

### CODE REVIEW

File: maven-modules/maven-build-lifecycle/src/test/java/com/baeldung/mavenlifecycle/CourseAppUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class CourseAppUnitTest {
+
+    @Mock
```
Comment (acerasoni) [inline maven-modules/maven-build-lifecycle/src/test/java/com/baeldung/mavenlifecycle/CourseAppUnitTest.java:14]: Why are we using a mock here but not in the integration test?

File: maven-modules/maven-build-lifecycle/src/test/java/com/baeldung/mavenlifecycle/CourseAppIT.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+class CourseAppIT {
+
+    @Test
+    void givenIntegrationTest_whenGetCourse_ThenCourseShouldBePresent() {
+        CourseApp courseApp = new CourseApp();
```
Comment (acerasoni) [inline maven-modules/maven-build-lifecycle/src/test/java/com/baeldung/mavenlifecycle/CourseAppIT.java:11]: In both test classes, can we add a line between setup and assertion within our tests? For example,

` void whenGetCourse_ThenCourseShouldBePresent() {
        when(courseApp.getCourse()).thenReturn("Baeldung Spring Masterclass");

        assertEquals("Baeldung Spring Masterclass", courseApp.getCourse());
    }`

File: maven-modules/maven-build-lifecycle/src/test/java/com/baeldung/mavenlifecycle/CourseAppUnitTest.java (hunk 1 above)
Comment (shahulbasha) [reply maven-modules/maven-build-lifecycle/src/test/java/com/baeldung/mavenlifecycle/CourseAppUnitTest.java]: this is addressed

File: maven-modules/maven-build-lifecycle/src/test/java/com/baeldung/mavenlifecycle/CourseAppIT.java (hunk 2 above)
Comment (shahulbasha) [reply maven-modules/maven-build-lifecycle/src/test/java/com/baeldung/mavenlifecycle/CourseAppIT.java:11]: this is addressed

File: maven-modules/pom.xml [hunk 3]
```diff
@@ -50,8 +50,10 @@
         <module>maven-pom-types</module>
         <module>multimodulemavenproject</module>
         <module>resume-from</module>
+        <module>maven-build-lifecycle</module>
```
Comment (KevinGilmore) [inline maven-modules/pom.xml:53]: Generally, the modules section should be in alphabetical order.

**Pull request closed without merging**